contracts/src/systems/actions.cairo

```diff
@@ -100,21 +100,45 @@ mod actions {
             let player: ContractAddress = get_caller_address();
 
             mancala_game.validate_move(player, selected_pit);
-            let (mut current_player, mut opponent) = mancala_game.get_players(world);
-            // Get seeds from the selected pit and validate it's not empty
-            let mut seeds = mancala_game.get_seeds(current_player, selected_pit);
-            if seeds == 0 {
-                panic!("Selected pit is empty. Choose another pit.");
-            }
-            mancala_game.clear_pit(ref current_player, selected_pit);
-            mancala_game
-                .distribute_seeds(ref current_player, ref opponent, ref seeds, selected_pit);
-            if mancala_game.is_game_finished(current_player, opponent) {
-                mancala_game.set_winner(current_player, opponent);
+
+            if mancala_game.status == GameStatus::InProgress {
+                let (mut current_player, mut opponent) = mancala_game.get_players(world);
+                // Get seeds from the selected pit and validate it's not empty
+                let mut seeds = mancala_game.get_seeds(current_player, selected_pit);
+                if seeds == 0 {
+                    panic!("Selected pit is empty. Choose another pit.");
+                }
+                mancala_game.clear_pit(ref current_player, selected_pit);
+                mancala_game
+                    .distribute_seeds(ref current_player, ref opponent, ref seeds, selected_pit);
+                if mancala_game.is_game_finished(current_player, opponent) {
+                    mancala_game.status = GameStatus::Finished;
+                    mancala_game.set_winner(current_player, opponent);
+                }
+                set!(world, (mancala_game, current_player, opponent));
+                // return the current player so client has ability to know
+                (mancala_game.current_player, mancala_game.status)
+            } else {
+                set!(world, (mancala_game));
```

Comment:
i added this `set!` macro here, not sure if its needed? do we need to call this macro whenever we modify elements of the world (player, mancala game, etc?)

Reply:
Yes the set!() will modify the state of the data on chain. Its needed anytime a model object is being updated.

Reply:
got it thanks!